include/secp256k1.h

```diff
@@ -80,6 +80,26 @@ typedef struct {
     unsigned char data[64];
 } secp256k1_ecdsa_signature;
 
+/** Data structure that holds a sign-to-contract ("s2c") opening. Sign-to-contract
+ *  allows a signer to commit to some data as part of a signature. It can be used as
+ *  an Out-argument in certain signing functions.
+ *
+ *  This structure is not opaque, but it is strongly discouraged to read or write to
+ *  it directly.
+ *
+ *  The exact representation of data inside is implementation defined and not
+ *  guaranteed to be portable between different platforms or versions. It can
+ *  be safely copied/moved.
+ */
+typedef struct {
+    /* magic is set during initialization */
+    unsigned char magic[8];
+    /* Public nonce before applying the sign-to-contract commitment */
+    secp256k1_pubkey original_pubnonce;
+    /* Integer indicating if signing algorithm negated the nonce */
+    unsigned char nonce_is_negated;
```

Comment:
Is this a computational optimization to avoid computing the jacobi symbol on `secp256k1_schnorrsig_verify_s2c_commit`?

Reply:
Yes, we could could compute the ec_commitment of `original_pubnonce` and `data` then negate if this is not a valid nonce. But this prevents batch verification. I will mention this in the doc.

Reply:
added comment to explain this